corgidrp/l2a_to_l2b.py

```diff
@@ -43,24 +44,28 @@ def dark_subtraction(input_dataset, dark_frame):
     darksub_dataset = input_dataset.copy()
 
     darksub_cube = darksub_dataset.all_data - dark_frame.data
-    
+
     # propagate the error of the dark frame
     if hasattr(dark_frame, "err"):
-        darksub_dataset.add_error_term(dark_frame.err[0], "dark_error")   
+        darksub_dataset.add_error_term(dark_frame.err[0], "dark_error")
     else:
         raise Warning("no error attribute in the dark frame")
-    
+
+    if hasattr(dark_frame, "dq"):
+        new_all_dq = darksub_dataset.all_dq + dark_frame.dq[0]
```

Comment:
It’s not immediately clear to me that we want to combine DQs between calibrations and data, but I haven’t thought it through really. Do you have an example of a case where we’d want that?

should we move forward with it, we’ll want to combine the DQs using numpy.bitwise_or

Reply:
This is the sum of the flag values (similar to how it is done here: https://github.com/roman-corgi/corgidrp/blob/e86d27a700bee3d47e5d42d78bbecd71507f63d6/corgidrp/l1_to_l2a.py#L216).  So in the end, this value can be done via bitwise_or, but the flag values (powers of 2) are added and stored in the dq.

Reply:
I think ultimately its most robust to use bitwise_or both here and in the cosmic ray code. Doing it as a sum puts assumptions on what's already in the all_dq file, and the bitwse_or does not.